nodes/basic-pow/src/service.rs

```diff
@@ -0,0 +1,189 @@
+//! Service and ServiceFactory implementation. Specialized wrapper over substrate service.
+
+use std::sync::Arc;
+use sc_client::LongestChain;
+use runtime::{self, GenesisConfig, opaque::Block, RuntimeApi};
+use sc_service::{error::{Error as ServiceError}, AbstractService, Configuration, ServiceBuilder};
+use sp_inherents::InherentDataProviders;
+use sc_executor::native_executor_instance;
+pub use sc_executor::NativeExecutor;
+use crate::pow::Sha3Algorithm;
+use sc_network::{config::DummyFinalityProofRequestBuilder};
+
+// Our native executor instance.
+native_executor_instance!(
+	pub Executor,
+	runtime::api::dispatch,
+	runtime::native_version,
+);
+
+pub fn build_inherent_data_providers() -> Result<InherentDataProviders, ServiceError> {
+	let providers = InherentDataProviders::new();
+
+	providers
+		.register_provider(sp_timestamp::InherentDataProvider)
+		.map_err(Into::into)
+		.map_err(sp_consensus::error::Error::InherentData)?;
+
+	// Attempt to register the babe inherent  because the super-runtime expects the babe inherent.
+	// See https://github.com/substrate-developer-hub/recipes/pull/173#issuecomment-599129425
+	// This hack didn't work: Checking inherents failed: timestamp set in block doesn't match slot in seal
+
+	// let bogus_babe_inherent = sp_consensus_babe::inherents::InherentDataProvider::new(1);
+	// providers
+	// 	.register_provider(bogus_babe_inherent)
+	// 	.map_err(Into::into)
+	// 	.map_err(sp_consensus::error::Error::InherentData)?;
```

Comment:
Ah, I see how it's connected. Yeah up to you if you think showing this is helpful to the overall point.

Reply:
I want to leave it there for the short-medium term until it's actually possible to do this sort of thing. It isn't mentioned in the write-up, and I think learners who look carefully enough at the code to notice it are ready for it.